core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_Team.hpp

```diff
@@ -0,0 +1,394 @@
+                    iType, Impl::OpenACCTeamMember>& loop_boundaries,
+                const Lambda& lambda, ReducerType& result) {
+  using ValueType = typename ReducerType::value_type;
+
+  ValueType tmp = ValueType();
+#pragma acc loop seq
+  for (iType i = loop_boundaries.start; i < loop_boundaries.end; i++)
+    lambda(i, tmp);
+  result = tmp;
+}
+
+// FIXME_OPENACC: custom reduction is not implemented.
+template <typename iType, class Lambda, typename ValueType, class JoinType>
+KOKKOS_INLINE_FUNCTION void parallel_reduce(
+    const Impl::TeamThreadRangeBoundariesStruct<iType, Impl::OpenACCTeamMember>&
+        loop_boundaries,
+    const Lambda& lambda, const JoinType& join, ValueType& init_result) {
+  static_assert(!Kokkos::Impl::always_true<Lambda>::value,
+                "custom reduction is not implemented");
+}
+
+// Hierarchical Parallelism -> Thread vector level implementation
+#pragma acc routine seq
+template <typename iType, class Lambda, typename ValueType>
+KOKKOS_INLINE_FUNCTION void parallel_reduce(
+    const Impl::ThreadVectorRangeBoundariesStruct<
+        iType, Impl::OpenACCTeamMember>& loop_boundaries,
+    const Lambda& lambda, ValueType& result) {
+  ValueType tmp = ValueType();
+
+#pragma acc loop seq
+  for (iType i = loop_boundaries.start; i < loop_boundaries.end; i++) {
+    lambda(i, tmp);
+  }
+  result = tmp;
+}
+
+#pragma acc routine seq
+template <typename iType, class Lambda, typename ReducerType>
+KOKKOS_INLINE_FUNCTION std::enable_if_t<Kokkos::is_reducer<ReducerType>::value>
+parallel_reduce(const Impl::ThreadVectorRangeBoundariesStruct<
+                    iType, Impl::OpenACCTeamMember>& loop_boundaries,
+                const Lambda& lambda, ReducerType const& result) {
+  using ValueType = typename ReducerType::value_type;
+
+  ValueType vector_reduce;
+  ReducerType::init(vector_reduce);
+
+#pragma acc loop seq
+  for (iType i = loop_boundaries.start; i < loop_boundaries.end; i++) {
+    lambda(i, vector_reduce);
+  }
+  result.reference() = vector_reduce;
+}
+
+// FIXME_OPENACC: custom reduction is not implemented.
+template <typename iType, class Lambda, typename ValueType, class JoinType>
+KOKKOS_INLINE_FUNCTION void parallel_reduce(
+    const Impl::ThreadVectorRangeBoundariesStruct<
+        iType, Impl::OpenACCTeamMember>& loop_boundaries,
+    const Lambda& lambda, const JoinType& join, ValueType& init_result) {
+  static_assert(!Kokkos::Impl::always_true<Lambda>::value,
+                "custom reduction is not implemented");
+}
+
+// Hierarchical Parallelism -> Team vector level implementation
+template <typename iType, class Lambda, typename ValueType>
+KOKKOS_INLINE_FUNCTION void parallel_reduce(
+    const Impl::TeamVectorRangeBoundariesStruct<iType, Impl::OpenACCTeamMember>&
+        loop_boundaries,
+    const Lambda& lambda, ValueType& result) {
+  ValueType tmp = ValueType();
+
+// Team-vector-level reduction is not supported in the OpenACC backend.
```

Comment:
Does that mean that this function will not do the right thing?

Reply:
That comment is wrong; deleted.